src/apps/statements/service/storeStatements/updateFullActivities.ts

```diff
@@ -43,79 +92,83 @@ export interface Opts {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.description !== undefined
-      ) ? matchingActivity.definition.description : {};
-    });
-    const extensions = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.extensions !== undefined
-      ) ? matchingActivity.definition.extensions : {};
-    });
-    const types = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.type !== undefined
-      ) ? matchingActivity.definition.type : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-    const moreInfos = matchingActivities.map((matchingActivity) => {
-      return (
-        matchingActivity.definition !== undefined &&
-        matchingActivity.definition.moreInfo !== undefined
-      ) ? matchingActivity.definition.moreInfo : undefined;
-    }).filter((value) => {
-      return value !== undefined;
-    });
-
-    return {
-      activityId,
-      name: Object.assign({}, ...names),
-      description: Object.assign({}, ...descriptions),
-      extensions: Object.assign({}, ...extensions),
-      ...(types.length > 0 ? { type: types[types.length - 1] } : {}),
-      ...(moreInfos.length > 0 ? { moreInfo: moreInfos[moreInfos.length - 1] } : {}),
-    };
-  });
-
-  await config.repo.updateFullActivities({ updates, client });
-  // tslint:disable-next-line:max-file-line-count
+  const groupedActivities = groupBy(
+    definedActivities,
+    (activity) => activity.activityId,
+  );
+
+  const fullActivities = map(
+    groupedActivities,
+    (matchingActivities, activityId): FullActivityDatabase => {
+      const names = matchingActivities.map(
+        (matchingActivity) => matchingActivity.name,
+      );
+
+      const descriptions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.description,
+      );
+
+      const extensions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.extensions,
+      );
+
+      const contextActivitiesForFullActivity: FullActivityContextActivities[] =
+        matchingActivities.map((matchingActivity) =>
+          mapValues(matchingActivity.context?.contextActivities, (matchingContextActivities) =>
+            matchingContextActivities !== undefined
+              // tslint:disable-next-line:max-line-length
+              ? matchingContextActivities.map((matchingContextActivity) => matchingContextActivity.id)
+              : matchingContextActivities,
+          ));
+
+      const mergedContextActivities = contextActivitiesForFullActivity.reduce(
+        (acc, contextActivity) => {
+
+          return mergeWith(acc, contextActivity, (objValue, srcValue) =>
+            isArray(objValue) ? uniq(objValue.concat(srcValue)) : objValue);
```

Comment:
@PrinceWaune for consistency with other changes in this commit (not sure about the whole code base, sorry), can we do line break structuring for readability?